### ISSUE TITLE
Different Namespaces in an OL event

### ISSUE THREAD

**Issue by MMoussabbih** [OpenLineage/OpenLineage#1776]
### Discussed in https://github.com/OpenLineage/OpenLineage/discussions/1775

<div type='discussions-op-text'>

<sup>Originally posted by **MMoussabbih** April 14, 2023</sup>
Hello everyone,

I'm using the **OpenLineage JAR** for **Spark integration** and I noticed that **namespace** used in `"inputs"` is not the same as what I configured in `spark.openlineage.namespace: "new-test"`, input dataset is created in `"file"` namespace. Here is the event : 

>
    "eventType": "COMPLETE",
    "eventTime": "2023-04-14T07:37:51.376Z",
    "run": {
      "runId": "f560e90b-131f-4d14-8b35-f3262b323995",
      "facets": {
        "spark.logicalPlan": {
          "_producer": "https://github.com/OpenLineage/OpenLineage/tree/0.22.0/integration/spark",
          "_schemaURL": "https://openlineage.io/spec/1-0-5/OpenLineage.json#/$defs/RunFacet",
          "plan": [
            {
              "class": "org.apache.spark.sql.catalyst.plans.logical.Aggregate",
              "num-children": 1,
              "groupingExpressions": [],
              "aggregateExpressions": [
                [
                  {
                    "class": "org.apache.spark.sql.catalyst.expressions.Alias",
                    "num-children": 1,
                    "child": 0,
                    "name": "count(1)",
                    "exprId": {
                      "product-class": "org.apache.spark.sql.catalyst.expressions.ExprId",
                      "id": 20,
                      "jvmId": "73a10800-6579-4921-9ddc-b4239fa78272"
                    },
                    "qualifier": [],
                    "nonInheritableMetadataKeys": []
                  },
                  {
                    "class": "org.apache.spark.sql.catalyst.expressions.aggregate.AggregateExpression",
                    "num-children": 1,
                    "aggregateFunction": 0,
                    "mode": {
                      "object": "org.apache.spark.sql.catalyst.expressions.aggregate.Complete$"
                    },
                    "isDistinct": false,
                    "resultId": {
                      "product-class": "org.apache.spark.sql.catalyst.expressions.ExprId",
                      "id": 15,
                      "jvmId": "73a10800-6579-4921-9ddc-b4239fa78272"
                    }
                  },
                  {
                    "class": "org.apache.spark.sql.catalyst.expressions.aggregate.Count",
                    "num-children": 1,
                    "children": [
                      0
                    ]
                  },
                  {
                    "class": "org.apache.spark.sql.catalyst.expressions.Literal",
                    "num-children": 0,
                    "value": "1",
                    "dataType": "integer"
                  }
                ]
              ],
              "child": 0
            },
            {
              "class": "org.apache.spark.sql.catalyst.plans.logical.Project",
              "num-children": 1,
              "projectList": [],
              "child": 0
            },
            {
              "class": "org.apache.spark.sql.execution.datasources.LogicalRelation",
              "num-children": 0,
              "relation": null,
              "output": [
                [
                  {
                    "class": "org.apache.spark.sql.catalyst.expressions.AttributeReference",
                    "num-children": 0,
                    "name": "NAME",
                    "dataType": "string",
                    "nullable": true,
                    "metadata": {},
                    "exprId": {
                      "product-class": "org.apache.spark.sql.catalyst.expressions.ExprId",
                      "id": 0,
                      "jvmId": "73a10800-6579-4921-9ddc-b4239fa78272"
                    },
                    "qualifier": []
                  }
                ],
                [
                  {
                    "class": "org.apache.spark.sql.catalyst.expressions.AttributeReference",
                    "num-children": 0,
                    "name": "ORIGIN_NAME",
                    "dataType": "string",
                    "nullable": true,
                    "metadata": {},
                    "exprId": {
                      "product-class": "org.apache.spark.sql.catalyst.expressions.ExprId",
                      "id": 1,
                      "jvmId": "73a10800-6579-4921-9ddc-b4239fa78272"
                    },
                    "qualifier": []
                  }
                ],
                [
                  {
                    "class": "org.apache.spark.sql.catalyst.expressions.AttributeReference",
                    "num-children": 0,
                    "name": "value",
                    "dataType": "long",
                    "nullable": true,
                    "metadata": {},
                    "exprId": {
                      "product-class": "org.apache.spark.sql.catalyst.expressions.ExprId",
                      "id": 2,
                      "jvmId": "73a10800-6579-4921-9ddc-b4239fa78272"
                    },
                    "qualifier": []
                  }
                ]
              ],
              "isStreaming": false
            }
          ]
        },
        "spark_version": {
          "_producer": "https://github.com/OpenLineage/OpenLineage/tree/0.22.0/integration/spark",
          "_schemaURL": "https://openlineage.io/spec/1-0-5/OpenLineage.json#/$defs/RunFacet",
          "spark-version": "3.3.0",
          "openlineage-spark-version": "0.22.0"
        }
      }
    },
    "job": {
      "namespace": "new-test",
      "name": "test.adaptive_spark_plan",
      "facets": {}
    },
    "inputs": [
      {
        "namespace": "file",
        "name": "/Users/mac-35535/Documents/data/inputs/testdata",
        "facets": {
          "dataSource": {
            "_producer": "https://github.com/OpenLineage/OpenLineage/tree/0.22.0/integration/spark",
            "_schemaURL": "https://openlineage.io/spec/facets/1-0-0/DatasourceDatasetFacet.json#/$defs/DatasourceDatasetFacet",
            "name": "file",
            "uri": "file"
          },
          "schema": {
            "_producer": "https://github.com/OpenLineage/OpenLineage/tree/0.22.0/integration/spark",
            "_schemaURL": "https://openlineage.io/spec/facets/1-0-0/SchemaDatasetFacet.json#/$defs/SchemaDatasetFacet",
            "fields": [
              {
                "name": "NAME",
                "type": "string"
              },
              {
                "name": "ORIGIN_NAME",
                "type": "string"
              },
              {
                "name": "value",
                "type": "long"
              }
            ]
          }
        },
        "inputFacets": {}
      }
    ],
    "outputs": [],
    "producer": "https://github.com/OpenLineage/OpenLineage/tree/0.22.0/integration/spark",
    "schemaURL": "https://openlineage.io/spec/1-0-5/OpenLineage.json#/$defs/RunEvent"

Do you have any idea about how we can have the **same namespace** for both `"job"` and `"inputs"` ?

Thanks in advance.</div>

**Comment by mobuchowski** [comment]
Hey @MMoussabbih - the name that you set is ment to describe job namespaces, which reflect the systems where data is processed. Dataset namespaces are ment to reflect different systems where the datasets are stored.

>Do you have any idea about how we can have the same namespace for both "job" and "inputs" ?

With OpenLineage integrations in this repo, you can't, as we have opinionated argument on what the dataset namespace is.

**Comment by MMoussabbih** [comment]
Hey @mobuchowski. Thanks for your answer.
In our case, we are using **AWS S3** as a **Datalake** component where all our datasets are stored. With this behaviour, we will have **hundreds** of **datasets namespaces** having as values the **S3 buckets names**, which are different to the `spark.openlineage.namespace: "new-test"`.
Do you have a better approach to solve this problem ?

**Comment by pawel-big-lebowski** [comment]
Hi @MMoussabbih,

As Maciej mentioned, the `spark.openlineage.namespace` setting sets the job namespace. However, we used to have a bug which set namespace `file` for datasets even if the namespace should be different.  It got resolved just today: https://github.com/OpenLineage/OpenLineage/pull/2142

If you store data in S3 bucket, namespace should NOT be `file`. However, so far we were not able to track the issue. 
I've even created some time ago an integration test for this https://github.com/OpenLineage/OpenLineage/blob/main/integration/spark/app/src/test/java/io/openlineage/spark/agent/lifecycle/SparkReadWriteIntegTest.java#L690 
which succeeded without helping me to find root cause.

Thanks for your patience. I hope we found solution for the problem you have.
